app/models/run.rb

```diff
@@ -103,4 +103,13 @@ def variables
   def env_variables
     Variable.to_hash(variables)
   end
+
+  # Called when a run has finished. Perform any post-run work here.
+  def finished!
+    return unless scraper.present?
```

Comment:
Maybe move this back down to `Morph::Runner`? It made me scratch my head when I first saw it. We're also doing another check for it down on the Runner anyway.